### PR TITLE
Site Editor: update headings hierarchy in the 'Manage all' screens

### LINKED ISSUE #52254: Site Editor: Incorrect headings hierarch in the 'Manage all' screens
### Description

Note: In the following screenshots, I'm using the HeadingsMap browser extension (available for Chrome and Firefox) to visually illustrate the headings in the page.

In  'Manage all templates',  'Manage all template parts', and possibly other 'manage all' screens, the headings hierarchy is incorrect and skips a few levels. A correct headings hierarchy is of fundamental importance for screen rearer users to efficiently navigate the content.

Edit: Please note there are also two H1 headings, which is incorrect.

I seem to recall the hierarchy was already fixed in a PR a while ago so I'm going to mark this issue as a regression.

On a side note: it would be great to re-focus all designers and developers attention to always check for a correct headings hierarchy. It's such a simple thing to implement but in the history of this project there have been several cases where new features have been shipped with a broken headings structure. @mtias I'd greatly appreciate any suggestions on the best way to train developers on this and make sure everybody is always focused on using headings correctly 🙏 
See also https://github.com/WordPress/gutenberg/issues/52253

This is basic semantics / accessibility / HTML. It would be great to just not use heading levels randomly.

ScreenshotL All templates:

Screenshot 2023-07-03 at 16 36 55

ScreenshotL All template parts:

Screenshot 2023-07-03 at 15 43 39

### Step-by-step reproduction instructions

- Use the HeadingsMap browser extension.
- Go to the Site editor > Design > Templates > Manage all templates.
- Observe the headings hierarchy is incorrect.
- Go to the Site editor > Design > Patterns > Manage all template parts.
- Observe the headings hierarchy is incorrect.

### Screenshots, screen recording, code snippet

_No response_

### Environment info

_No response_

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

### PR DESCRIPTION
**Pull request by ramonjd**
## What?
Resolves https://github.com/WordPress/gutenberg/issues/52254

This PR:

- updates heading hierarchy on the template and template part pages
- passes a `level` prop to Header from Page component

## Why?
The heading hierarchy was incorrect in that there were two h1s and, underneath, multiple h5s. There should be a incremental order.

## Testing Instructions

Taken from https://github.com/WordPress/gutenberg/issues/52254

1. Use the HeadingsMap browser extension.
2. Go to the **Site editor > Design > Templates > Manage all templates**.
3. Check that the headings hierarchy is correct (h1 > h2 > h3).
4. Go to the **Site editor > Design > Patterns > Manage all template parts**.
5. Observe the headings  hierarchy is correct (h1 > h2 > h3).

Compare with trunk - the font size should be pretty close.

## Screenshots or screencast 

Screenshot 2023-07-05 at 9 04 29 am
Screenshot 2023-07-05 at 9 04 17 am
Screenshot 2023-07-05 at 9 04 04 am

### CODE REVIEW

File: packages/edit-site/src/components/layout/style.scss [hunk 1]
```diff
@@ -313,5 +313,14 @@
...
 	}
+}
 
+.edit-site-page-template-parts,
+.edit-site-page-templates {
+	h2 {
+		font-size: 0.9rem;
+	}
+	h3 {
+		font-size: 0.82rem;
```
Comment (ntsekouras) [inline packages/edit-site/src/components/layout/style.scss]: These values seem a bit too specific, where they feel a bit arbitrary. I'd defer to @WordPress/gutenberg-design for that 😄

**Review by ntsekouras — APPROVED** [review]
This LGTM code wise, thanks! I've left a small comment about the font sizes.

File: packages/edit-site/src/components/layout/style.scss (hunk 1 above)
Comment (ramonjd) [reply packages/edit-site/src/components/layout/style.scss]: Thanks for looking! I was trying to match what was there before, which, if I had to give an opinion, were a tad too small anyway.

Comment (afercia) [reply packages/edit-site/src/components/layout/style.scss]: To my understanding, based on how the code currently works, a way to render a H2 and still get the H5 default font size would be this:

`<Heading as="h2" level={ 5 }>`

In this case, the `level` prop would only determine the font size.
I find this pretty confusing, at the point I'd consider to reinstate the `size` prop on the Heading component. Cc @ciampo 

IMHO, a way more intuitive syntax would be:

`<Heading level={ 2 } size={ 5 >`

To my understanding,  would also better follow the logic in [getHeadingFontSize](https://github.com/WordPress/gutenberg/blob/5ba748472b3e481e89ec3ac4c4e215031cc1eb3c/packages/components/src/ui/utils/font-size.ts#L62-L69), which checks whether the passed value is included or not within the 1-6 values. 

In other words, `level` should only allow values between 1 and 6 and map to the default heading font sizes. Instead, `size` should allow any arbitrary font size value.

Right now, it is possible to pass something like `level={ 48 }` and actually get a 48px font size. That seems misleading to me, as the prop name `level` clearly hints at the headings 1-6 levels.

Will create a separate issue.

Comment (ramonjd) [reply packages/edit-site/src/components/layout/style.scss]: > <Heading level={ 2 } size={ 5 >

That makes sense, thanks. I can play around with it.

Comment (jameskoster) [reply packages/edit-site/src/components/layout/style.scss]: +1, we should try to avoid custom sizing wherever possible.

Comment (ramonjd) [reply packages/edit-site/src/components/layout/style.scss]: I rushed the first commit. I just changed the `<Heading />` props, removed the CSS and I think it's good for another check.

**Comment by tellthemachines** [comment]
I just cherry-picked this PR to the update/first-pre-beta-4 branch to get it included in the next release: 8dab0a04a1
